opm/simulators/flow/RSTConv.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+  Copyright 2023 Equinor ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <config.h>
+#include <opm/simulators/flow/RSTConv.hpp>
+
+#include <opm/input/eclipse/Schedule/Schedule.hpp>
+#include <opm/input/eclipse/Schedule/RSTConfig.hpp>
+
+#include <dune/common/fvector.hh>
+#include <dune/istl/bvector.hh>
+
+#include <algorithm>
+#include <cmath>
+
+namespace Opm {
+
+void RSTConv::init(const std::size_t numCells,
+                   const RSTConfig& rst_config,
+                   const std::array<int,3>& compIdx)
+{
+    const auto kw = rst_config.keywords.find("CONV");
+    if (kw == rst_config.keywords.end()) {
+        N_ = 0;
+        cnv_X_.clear();
+        return;
+    }
+
+    N_ = kw->second;
+    compIdx_ = compIdx;
+
+    cnv_X_.resize(3);
+    for (std::size_t i = 0; i < 3; ++i) {
+        if (compIdx_[i] != -1) {
+            cnv_X_[i].resize(numCells);
+        }
+    }
+}
+
+template<class ResidualVector>
+void RSTConv::update(const ResidualVector& resid)
+{
+    if (N_ == 0) {
+        return;
+    }
+
+    std::vector<int> cix(resid.size());
+    std::iota(cix.begin(), cix.end(), 0);
+    for (std::size_t i = 0; i < cnv_X_.size(); ++i) {
+        if (cnv_X_[i].empty()) {
+            continue;
+        }
+        std::partial_sort(cix.begin(), cix.begin() + N_, cix.end(),
+                          [&resid,&i,this](const int a, const int b)
+                          { return std::abs(resid[a][compIdx_[i]]) > std::abs(resid[b][compIdx_[i]]); });
```

Comment:
You could probably capture
```C++
[&resid, c = this->compIdx_[i]](const int a, const int b)
```
instead and then refer to `resid[a][c]` and `resid[b][c]` in the body of the lambda instead of capturing `this` and `i`.

Reply:
sure.